File: android/live-update-provider/src/main/kotlin/io/ionic/liveupdateprovider/LiveUpdateProvider.kt
package io.ionic.liveupdateprovider

import android.content.Context

/** Creates managers for a live update provider. */
interface LiveUpdateProvider {
    /**
     * Creates a manager for a single app.
     *
     * @param context Android context for manager creation.
     * @param configuration Provider-specific configuration values.
     * @throws ProviderError.InvalidConfiguration when [configuration] is invalid.
     * @return A manager that can sync and resolve the app's latest files.
     */
    @Throws(ProviderError.InvalidConfiguration::class)
    fun createManager(
        context: Context,
        configuration: Map<String, Any>
    ): ProviderManager
}

File: ios/Sources/LiveUpdateProvider/LiveUpdateProvider.swift
/// Creates managers for a live update provider.
public protocol LiveUpdateProvider {
    /// Creates a manager for a single app.
    ///
    /// - Parameter configuration: Provider-specific configuration values.
    /// - Throws: `ProviderError.invalidConfiguration` when the configuration is invalid.
    /// - Returns: A manager that can sync and resolve the app's latest files.
    func createManager(configuration: [String: Any]) throws -> any ProviderManager
}
